File: example/src/main/java/ru/virgil/example/image/FileTypeService.java
package ru.virgil.example.image;

import org.apache.tika.Tika;
import org.springframework.stereotype.Service;

@Service
public class FileTypeService extends Tika {

    // todo: в утилиты
    public String checkIfImage(byte[] content) {
        String mimeType = detect(content);
        String imageTypeRegex = "image/.*";
        if (!mimeType.matches(imageTypeRegex)) {
            throw new UnsupportedOperationException("It should be an image (%s). Given file: %s"
                    .formatted(imageTypeRegex, mimeType));
        }
        return mimeType;
    }

}

File: tools/src/main/kotlin/ru/virgil/spring_tools/tools/image/FileTypeService.kt
package ru.virgil.spring_tools.tools.image

import org.apache.tika.Tika
import org.springframework.stereotype.Component

@Component
open class FileTypeService : Tika() {

    fun getImageMimeType(content: ByteArray): String {
        val mimeType = detect(content)
        val imageTypeRegex = "image/.*"
        if (!mimeType.matches(imageTypeRegex.toRegex())) {
            throw UnsupportedOperationException("File should be an image ($imageTypeRegex). Given file: $mimeType")
        }
        return mimeType
    }
}